### PR TITLE
Add MySql support

### PR DESCRIPTION
**Pull request by adlamas**
- Update adapter helper to support MySQL json types
- Update generator spec to permit using MySql
- Update migration spec
- Update CI to check MySql's support

### CODE REVIEW

File: spec/generators/active_outbox_generator_spec.rb [hunk 1]
```diff
@@ -61,7 +61,7 @@ def change
               create_table :#{table_name}_outboxes do |t|
                 t.string :identifier, null: false, index: { unique: true }
                 t.string :event, null: false
-                t.string :payload
```
Comment (guillermoap) [inline spec/generators/active_outbox_generator_spec.rb:64]: Add a specific MySQL test, and change the name for this one to be 'when it is a sqlite migration'

File: lib/active_outbox/adapter_helper.rb [hunk 2]
```diff
@@ -3,15 +3,25 @@
 module ActiveOutbox
   module AdapterHelper
     def self.uuid_type
-      postgres? ? 'uuid' : 'string'
+      return 'uuid' if postgres?
```
Comment (guillermoap) [inline lib/active_outbox/adapter_helper.rb:6]: can we use case? I feel it's cleaner that returns

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -48,4 +62,17 @@ jobs:
           ADAPTER: postgresql
       - name: Run RuboCop
         run: bundle exec rubocop .
-
+      - name: install mysql client
```
Comment (guillermoap) [inline .github/workflows/ci.yml:65]: speaking out of ignorance, why is the client needed?

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -27,6 +27,20 @@ jobs:
           --health-interval 10s
           --health-timeout 5s
           --health-retries 5
+      mysql:
+        image: mysql
+        ports:
+          - 8888:3306
```
Comment (guillermoap) [inline .github/workflows/ci.yml:33]: why not use 3306?

**Review by guillermoap — CHANGES_REQUESTED** [review]
Requesting changes mainly because of the missing spec. Also, can you cleanup the git history? Looking great 💪🏻

**Comment by guillermoap** [comment]
Also remember to bump version, and the run a bundle install

File: .github/workflows/ci.yml (hunk 3 above)
Comment (adlamas) [reply .github/workflows/ci.yml]: I removed it. When I was figuring out how to make it work, I found a couple of articles saying that you need to install MySQL client to use it on CI.
Looks like it was fixed, so I removed it

File: .github/workflows/ci.yml (hunk 4 above)
Comment (adlamas) [reply .github/workflows/ci.yml]: I updated it. Looked like GH actions didn't permit using this port for Mysql Docker images if the default MySql process wasn't shutting down, but looks like it was changed too.

**Review by guillermoap — APPROVED** [review]

**Pull request merged**